crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1131,17 +1131,20 @@ impl PgIndexerStore {
                         0 | 1 => 0,
                         _ => {
                             let prev_epoch_id = epoch_id - 2;
-                            let result = checkpoints::table
+                            let last_epoch_max_tx_sequence_number = checkpoints::table
                                 .filter(checkpoints::epoch.eq(prev_epoch_id as i64))
-                                .select(max(checkpoints::network_total_transactions))
+                                .select(checkpoints::max_tx_sequence_number)
+                                .order_by(checkpoints::sequence_number.desc())
                                 .first::<Option<i64>>(conn)
-                                .await
-                                .map(|o| o.unwrap_or(0))?;
-
-                            result as u64
+                                .await?
+                                .ok_or_else(|| {
+                                    IndexerError::PersistentStorageDataCorruptionError(
+                                        "max_tx_sequence_number should not be None".to_string(),
+                                    )
+                                })?;
+                            (last_epoch_max_tx_sequence_number as u64).saturating_add(1)
```

Comment:
having var `last_epoch_max_tx_sequence_number` to make conversion from `last_epoch_max_tx_sequence_number` to `network_total_transactions` a bit more obvious

Reply:
Do you still need the `+1` here?